internal/controller/standalone_pgadmin/pod.go

```diff
@@ -127,19 +148,51 @@ func pod(
 				MountPath: "/var/log/pgadmin",
 			},
 			{
-				Name:      "tmp",
+				Name:      mountedConfigsVolumeName,
+				MountPath: configMountPath + "/conf.d",
+				ReadOnly:  true,
+			},
+			{
+				Name:      tempVolumeName,
 				MountPath: "/tmp",
 			},
 		},
 	}
+	startup := corev1.Container{
+		Name:            naming.ContainerPGAdminStartup,
+		Command:         startupCommand(),
+		Image:           container.Image,
+		ImagePullPolicy: container.ImagePullPolicy,
+		Resources:       container.Resources,
+		SecurityContext: initialize.RestrictedSecurityContext(),
+		VolumeMounts: []corev1.VolumeMount{
+			// Volume to write a custom `config_system.py`` file to.
+			{
+				Name:      configSystemVolumeName,
+				MountPath: configMountPath,
+				ReadOnly:  false,
+			},
+			// Volume mounted here because of related mount path.
+			// If volume is not mounted here, K8s errors out trying to create
+			// initContainer and container with different read-only values at related mount path
+			{
+				Name:      mountedConfigsVolumeName,
+				MountPath: configMountPath + "/conf.d",
+				ReadOnly:  true,
+			},
```

Comment:
❓ I'm not grokking this comment. Can you explain it to me?

I don't see `startupCommand()` reading from this directory. Perhaps we can omit the mount completely?

Reply:
So here's what I was thinking and wanted to bring up:

1. "we're not using that `etc/pgadmin/conf.d` in the initContainer, so let's drop it" -- that led to an error with `read-only file system: unknown`
2. "hmmm, ok, well we've got one vol at `etc/pgadmin` and one _inside_ it at `etc/pgadmin/conf.d` -- what happens if I drop the `etc/pgadmin/conf.d` from the initContainer and instead mount it at the container at `var/pgadmin`?" -- and that seemed to work.

So it seemed like defining the parent volume (`etc/pgadmin`) as `rw` in the initcontainer and then the parent + child vols as `ro` in the container led to some confusion.

Not sure if that description makes sense.

Reply:
We've got a few ways to deal with this, which I'd like to present from worst to best:

1. leave `/etc/pgadmin` rw in the main container -- yikes, we don't want things there because they can run
2. have the initContainer `mkdir etc/pgadmin/conf.d` and not mount the volume there -- it works, but it runs code
3. leave the unused `etc/pgadmin/conf.d` volume mounted to the initcontainer -- it's unused, but mounting an unused vol to an init (therefore temporary) container doesn't seem _wasteful_

Probably some other way to handle this, but with some experimentation, that's what I've got.

Reply:
I've looked into it, and here's what's happening:

1. Kubernetes (kubelet) preps a bunch of directories to act as the emptyDir, ConfigMap, and Secret volumes.
2. It passes a list of bind mounts (directories outside the container paired with paths inside the container) to runc during container start.
3. runc then mounts them in sequence, calling mkdir on the path within the container chroot, "rootfs."

It's here that runc finds the higher directory is readonly (via mount), and it cannot mkdir the subdirectory.

This does _not_ happen when the higher directory is a ConfigMap or Secret because Kubernetes does those mkdir during step 1: https://pr.k8s.io/60342

I've opened https://issue.k8s.io/121294 to see if Kubernetes may ever do this for us. In the meantime, we can explain what we do is a workaround for a known issue. 😁 

<br/>

> 2\. have the initContainer `mkdir etc/pgadmin/conf.d` and not mount the volume there -- it works, but it runs code

Now that I've looked into it, I like this option. This is exactly how Kubernetes does it, and "prepping the file system" in an init container seems fitting.